ledger/complete/compactor_test.go

```diff
@@ -479,6 +481,95 @@ func TestCompactorAccuracy(t *testing.T) {
 	})
 }
 
+// TestCompactorTriggeredByAdminTool tests that the compactor will listen to the signal from admin tool
+// to trigger checkpoint when current segment file is finished.
+func TestCompactorTriggeredByAdminTool(t *testing.T) {
+
+	const (
+		numInsPerStep      = 2 // the number of payloads in each trie update
+		pathByteSize       = 32
+		minPayloadByteSize = 2<<11 - 256 // 3840 bytes
+		maxPayloadByteSize = 2 << 11     // 4096 bytes
+		size               = 20
+		checkpointDistance = 5   // create checkpoint on every 5 segment files
+		checkpointsToKeep  = 0   // keep all
+		forestCapacity     = 500 // the number of tries to be included in a checkpoint file
+	)
+
+	metricsCollector := &metrics.NoopCollector{}
+
+	unittest.RunWithTempDir(t, func(dir string) {
+
+		rootHash := trie.EmptyTrieRootHash()
+
+		// Create DiskWAL and Ledger repeatedly to test rebuilding ledger state at restart.
+
+		wal, err := realWAL.NewDiskWAL(unittest.LoggerWithName("wal"), nil, metrics.NewNoopCollector(), dir, forestCapacity, pathByteSize, 32*1024)
+		require.NoError(t, err)
+
+		l, err := NewLedger(wal, forestCapacity, metricsCollector, unittest.LoggerWithName("ledger"), DefaultPathFinderVersion)
+		require.NoError(t, err)
+
+		compactor, err := NewCompactor(l, wal, unittest.LoggerWithName("compactor"), forestCapacity, checkpointDistance, checkpointsToKeep, atomic.NewBool(true))
+		require.NoError(t, err)
+
+		fmt.Println("should stop as soon as segment 5 is generated, which should trigger checkpoint 5 to be created")
+		fmt.Println("note checkpoint 0 will be be notified to the CompactorObserver because checkpoint 0 is the root checkpoint")
+		fmt.Println("Why fromBound =5? because without forcing checkpoint to trigger, it won't trigger until segment 4 is finished, with ")
+		fmt.Println("forcing checkpoint to trigger, it will trigger when segment 0 and 5 is finished")
+		fromBound := 5
+
+		co := CompactorObserver{fromBound: fromBound, done: make(chan struct{})}
+		compactor.Subscribe(&co)
+
+		// Run Compactor in background.
+		<-compactor.Ready()
+
+		fmt.Println("generate the tree and create WAL")
+		fmt.Println("13 segments will trigger checkpoint 5")
```

Comment:
```suggestion
		fmt.Println("6 segments will trigger checkpoint 5")
```

At least 6 segments (from segment 0 to 5) would be needed to trigger checkpoint 5 but there probably wouldn't be 13 segments created from the `for` loop because I think there would be around 1-2 records per segment using the random ranges specified in the test.

Random ranges specified in the test:
```
minPayloadByteSize = 2<<11 - 256 // 3840 bytes
maxPayloadByteSize = 2 << 11     // 4096 bytes
```